PhpStorm Protocol (Win)/run_editor.js

```diff
@@ -51,89 +32,58 @@ if (match) {
+                        if (fileSystemObj.FileExists(folderObject.Path + '\\bin\\phpstorm64.exe')) {
+                            process_name = 'phpstorm64.exe';
+                        }
 
-    var maxMajor = 0,
-        maxMinor = 0,
-        maxPatch = 0,
-        maxVersionFolder = "";
-    // Traverse through the fileCollection using the FOR loop
-    // read the maximum version from toolbox filesystem
-    for (var objEnum = new Enumerator(fileCollection); !objEnum.atEnd(); objEnum.moveNext()) {
-        var folderObject = ( objEnum.item() );
+                        var editor = folderObject.Path + '\\bin\\' + process_name;
 
-        if (folderObject.Name.lastIndexOf('plugins') !== -1) {
-            continue;
+                        dateLastModified = lastModifiedTime;
+                    }
+                }
+            }
         }
 
-        var versionMatch = /(\d+)\.(\d+)\.(\d+)/.exec(folderObject.Name),
-            major = parseInt(versionMatch[ 1 ]),
-            minor = parseInt(versionMatch[ 2 ]),
-            patch = parseInt(versionMatch[ 3 ]);
 
-        if (maxMajor === 0 || maxMajor <= major) {
-            if (maxMajor < major) {
-                maxMinor = 0;
-                maxPatch = 0;
-            }
-            maxMajor = major;
+    if (project !== '') {
+        editor += ' "%project%"';
+    }
 
-            if (maxMinor === 0 || maxMinor <= minor) {
-                if (maxMinor < minor) {
-                    maxPatch = 0;
-                }
-                maxMinor = minor;
+    editor += ' --line %line% "%file%"';
 
-                if (maxPatch === 0 || maxPatch <= patch) {
-                    maxPatch = patch;
-                    maxVersionFolder = folderObject.Name;
-                }
-            }
-        }
-    }
+    var command = editor.replace(/%line%/g, match[3])
+        .replace(/%file%/g, file)
+        .replace(/%project%/g, project)
+        .replace(/\//g, '\\');
 
-    settings.folder_name = maxVersionFolder;
+    shell.Exec(command);
 
-    // read version name and product name from product-info.json
-    var versionFile = fso.OpenTextFile(toolboxDirectory + settings.folder_name + "\\product-info.json", 1, true);
-    var content = versionFile.ReadAll();
+    var locator     = new ActiveXObject('WbemScripting.SWbemLocator'),
+        service     = locator.ConnectServer('.', '/root/CIMV2'),
+        processes   = service.ExecQuery('Select * from Win32_Process WHERE Name = "' + process_name + '"');
+
+        // focus on editor based process id
+        shell.AppActivate((new Enumerator(processes)).item(0).ProcessId);
+}
 
-    eval('var productVersion = ' + content + ';');
-    settings.window_title = 'PhpStorm ' + productVersion.version;
-    editor = '"' + toolboxDirectory + settings.folder_name + '\\' + productVersion.launch[ 0 ].launcherPath.replace(/\//g, '\\') + '"';
+function debug(text) {
+    return (new ActiveXObject('WScript.Shell')).Popup(text, 0, 'Debug', 48);
 }
```

Comment:
Is debug code really needed?

Reply:
No, but I prefer to keep it for as does. For debug proposes..